dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -96,20 +99,28 @@ public FileProvisioning(String configFile) {
             this.plugins = pluginManager.getExtensions(ProvisionInterface.class);
             this.preProvisionPlugins = pluginManager.getExtensions(PreProvisionInterface.class);
 
+            // Map of ProvisionInterface & PreProvisionInterface plugins
+            this.pluginsMap = this.plugins
+                    .stream()
+                    .collect(Collectors.toMap(plugin -> plugin.getClass().getName().split("\\$")[0], plugin -> plugin));
```

Comment:
Can you create a test for the logic. There already is a test file, FileProvisionTest that you could put the tests in.